### DISCUSSION TITLE
Motion detection grid-based regions to improve FP ratio

### DISCUSSION THREAD

**Discussion by marcinos** [Motion-Project/motion#1503] [General]
Hello, great product, I'm a long time user and supporter :)

As far as I understand it currently motion takes two pictures and compares them to determine if a difference exceeds predefined threshold. With that being said in a windy weather I can exceed my 4% threshold with 1% difference in the bottom left corner (plastic bag), 2.5% difference in the top right corner (trees) and 1% difference in the bottom right corner (fallen leaves moved by the wind), however a single person entering a still scene might not cause 4% threshold to be exceeded.

Would it make sense to divide the scene into 2, 4, 8 or even more grid-based regions and determine movement in each one of them separately? Thresholds can be then multiplied by regions count, so for 4 regions the actual threshold is going to be 16% per region. In my opinion more regions should lower the false positives ratio as unrelated events from different areas of the scene would not combine into overall motion detection. On the other hand one significant change in one of the regions would trigger a motion detection where currently it may need more contributing events from other regions

**Comment by erew123** [comment]
Im in agreement with the above and I too have issues where one of the existing editable mask is looking at distant objects e.g. a person on my grid, may only take up one of the grid squares and doesn't trigger the motion detection, because there isnt enough overall pixel change.

I would like to add another suggestion too though. This would be profiles for the settings. I have noticed that at night, I need to change my sensitivity levels for outdoor scenes on one of my cameras... so it would be great to be able to set 2 or more settings profiles, one that is applied in daytime and one at night time.... I guess this could be done either with a setting that knew your location by GPS coordinates, the time of day and sun up/sun down OR a more simple method, might be to have a grid, similar to the editable mask, and one of those squares, you could say "when this square goes below X brightness for X amount of minutes, load profile A, and when it goes above X brightness, load profile B".

Perhaps along with this change, you could also allow user settings to enhance the image being streamed at night e.g. contrast/brightness increases etc. (not sure how processor heavy that would be mind)

Thanks

**Comment by jdmaudio** [comment]
I think the grid approach would be a good thing to implement. I believe this is the approach used in opencv / gstreamer motioncells
https://gstreamer.freedesktop.org/documentation/opencv/motioncells.html?gi-language=c
